### PR TITLE
Shotgun cleanup

### PR DESCRIPTION
**Pull request by NinjaDanz3r**
Just general fixes and doodling just about everywhere.
Main concerns addressed:
Sonar issues.
Unused includes.
Unused functions.

### CODE REVIEW

File: src/Video/Lighting/Lighting.cpp [hunk 1]
```diff
@@ -79,15 +79,14 @@ void Lighting::Render(const glm::mat4& inverseProjectionMatrix, RenderSurface* r
     //glUniform1i(shaderProgram->GetUniformLocation("textureMetallic"), 2);
     //glUniform1i(shaderProgram->GetUniformLocation("textureRougness"), 3);
     //glUniform1i(shaderProgram->GetUniformLocation("tDepth"), 4);
-    //
+
```
Comment (NinjaDanz3r) [inline src/Video/Lighting/Lighting.cpp]: INFO comma should be followed by whitespace [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)

File: src/Video/FrameBuffer.cpp [hunk 2]
```diff
@@ -6,9 +6,7 @@
 
 using namespace Video;
 
-FrameBuffer::FrameBuffer(const std::vector<ReadWriteTexture*>& textures) {
-    this->textures = textures;
-
+FrameBuffer::FrameBuffer(const std::vector<ReadWriteTexture*>& textures) : textures(textures) {
     // Frame buffer object.
```
Comment (NinjaDanz3r) [inline src/Video/FrameBuffer.cpp]: MINOR Variable 'textures' is assigned in constructor body. Consider performing initialization in initialization list. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuseInitializationList)

File: src/Engine/Physics/Shape.hpp [hunk 3]
```diff
@@ -67,9 +67,10 @@ namespace Physics {
             /**
              * @return The Bullet shape.
              */
-            btCollisionShape* GetShape();
+            btCollisionShape* GetShape() const;
+
+            Shape(const Shape& other) = delete;
```
Comment (NinjaDanz3r) [inline src/Engine/Physics/Shape.hpp]: MINOR Class 'Shape' is unsafe, 'Shape::shape' can leak by wrong usage. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunsafeClassCanLeak)

File: src/Engine/Physics/Trigger.cpp [hunk 4]
```diff
@@ -25,7 +24,7 @@ namespace Physics {
         return 0; // Was a planned purpose, but is not used.
     }
 
-    btCollisionObject* Trigger::GetCollisionObject() {
+    btCollisionObject* Trigger::GetCollisionObject() const {
         return trigger;
```
Comment (NinjaDanz3r) [inline src/Engine/Physics/Trigger.cpp]: MINOR The function 'GetCollisionObject' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

File: src/Video/Renderer.cpp [hunk 5]
```diff
@@ -27,7 +27,7 @@ using namespace Video;
 Renderer::Renderer() {
```
Comment (NinjaDanz3r) [inline src/Video/Renderer.cpp]: MINOR Member variable 'Renderer::skinRenderProgram' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'Renderer::colorFilter' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'Renderer::fogFilter' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'Renderer::fxaaFilter' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'Renderer::glowFilter' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'Renderer::glowBlurFilter' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

File: src/Editor/main.cpp [hunk 6]
```diff
@@ -24,9 +24,12 @@
 
 int main() {
```
Comment (NinjaDanz3r) [inline src/Editor/main.cpp]: MINOR Return value of function freopen() is not used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AignoredReturnValue)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 29 issues
* ![MINOR][MINOR] 28 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L172](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Hymn.cpp#L172): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ParticleManager.cpp#L61](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/ParticleManager.cpp#L61): The function 'UpdateBuffer' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L373](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/RenderManager.cpp#L373): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L383](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/RenderManager.cpp#L383): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L401](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/RenderManager.cpp#L401): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L440](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/RenderManager.cpp#L440): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L458](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/RenderManager.cpp#L458): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L478](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/RenderManager.cpp#L478): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L499](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b6fca30315bb428c506fc76e9a11803651e57c23/src/Engine/Manager/RenderManager.cpp#L499): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/main.cpp [hunk 7]
```diff
@@ -24,9 +24,12 @@
...
     // Enable logging if requested.
-    if (EditorSettings::GetInstance().GetBool("Logging"))
-        freopen(FileSystem::DataPath("Hymn to Beauty", "log.txt").c_str(), "a", stderr);
-    
+    if (EditorSettings::GetInstance().GetBool("Logging")){
+        FILE* file = freopen(FileSystem::DataPath("Hymn to Beauty", "log.txt").c_str(), "a", stderr);
+        if (file == nullptr)
+            Log() << "Could not open logging file!";
```
Comment (Chainsawkitten) [inline src/Editor/main.cpp]: Add a newline at the end of this log.

File: src/Engine/Util/MousePicking.hpp [hunk 8]
```diff
@@ -48,14 +48,14 @@ class MousePicking {
         Entity* camera;
 
         // Return a vector
-        glm::vec3 CalculateRay();
+        glm::vec3 CalculateRay() const;
 
         // Return a vector with mouse coordinates in Normalized Device Coordinates.
-        glm::vec2 GetNDC(double mouseX, double mouseY);
+    static glm::vec2 GetNDC(double mouseX, double mouseY);
```
Comment (Chainsawkitten) [inline src/Engine/Util/MousePicking.hpp]: Incorrect indentation.

File: src/Engine/Manager/ResourceManager.hpp [hunk 9]
```diff
@@ -24,30 +24,9 @@ class ResourceManager {
...
-        Geometry::Cube* CreateCube();
-        
-        /// Free the reference to the cube.
-        /**
-         * Deletes the instance if no more references exist.
-         */
-        void FreeCube();
-        
+        /// Constructor
+        ResourceManager() {}
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ResourceManager.hpp]: MINOR Member variable 'ResourceManager::rectangle' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'ResourceManager::cube' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

File: src/Engine/Physics/Shape.hpp [hunk 10]
```diff
@@ -67,9 +67,10 @@ namespace Physics {
...
 
-        private:
             btCollisionShape* shape;
```
Comment (NinjaDanz3r) [inline src/Engine/Physics/Shape.hpp]: MINOR Class 'Shape' is unsafe, 'Shape::shape' can leak by wrong usage. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunsafeClassCanLeak)

File: src/Engine/Physics/Trigger.cpp [hunk 11]
```diff
@@ -25,7 +24,7 @@ namespace Physics {
         return 0; // Was a planned purpose, but is not used.
     }
 
-    btCollisionObject* Trigger::GetCollisionObject() {
+    btCollisionObject* Trigger::GetCollisionObject() const {
```
Comment (NinjaDanz3r) [inline src/Engine/Physics/Trigger.cpp]: MINOR The function 'GetCollisionObject' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 28 issues
* ![MINOR][MINOR] 28 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [main.cpp#L47](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Editor/main.cpp#L47): Variable 'vrSystem' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [Hymn.cpp#L175](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Hymn.cpp#L175): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ParticleManager.cpp#L59](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/ParticleManager.cpp#L59): The function 'UpdateBuffer' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L313](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L313): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L323](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L323): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L341](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L341): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L380](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L380): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L398](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L398): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L418](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L418): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 28 issues
* ![MINOR][MINOR] 28 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [main.cpp#L46](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Editor/main.cpp#L46): Variable 'vrSystem' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [Hymn.cpp#L175](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Hymn.cpp#L175): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ParticleManager.cpp#L59](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/ParticleManager.cpp#L59): The function 'UpdateBuffer' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L313](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L313): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L323](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L323): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L341](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L341): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L380](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L380): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L398](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L398): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L418](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/86d9aa357fbb06e759099d5ec32c45d20290b1c6/src/Engine/Manager/RenderManager.cpp#L418): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**